staging/src/k8s.io/client-go/util/cert/cert.go

```diff
@@ -57,8 +58,12 @@ type AltNames struct {
 // NewSelfSignedCACert creates a CA certificate
 func NewSelfSignedCACert(cfg Config, key crypto.Signer) (*x509.Certificate, error) {
 	now := time.Now()
+	serial, err := cryptorand.Int(cryptorand.Reader, new(big.Int).SetInt64(math.MaxInt64))
```

Comment:
actually, can't this still return zero?

> Int returns a uniform random value in [0, max). It panics if max <= 0.



/hold

Reply:
Ah fun!

@liggitt this snippet was copied from elsewhere in our repo :)

https://cs.k8s.io/?q=serial%2C%20err%20%3A%3D.*rand&i=nope&files=&excludeFiles=&repos=kubernetes/kubernetes

Reply:
we could fix the existing problem as part of this PR in a separate commit?

Reply:
Could someone help take a look at the update? if it is good, I will fix the existing problem in a separate commit with similar code.